scripts/rebase.sh

```diff
@@ -92,6 +95,34 @@ download_release() {
 }
 
 
+# Greps a Golang pseudoversion from input.
+grep_pseudoversion() {
+    local line=$1
+
+    echo "${line}" | grep -Po "v[0-9]+\.(0\.0-|\d+\.\d+-([^+]*\.)?0\.)\d{14}-[A-Za-z0-9]+(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?"
+}
+
+# Updates a replace directive using an embedded component's commit.
+# Caches component pseudoversions for faster processing.
+declare -A pseudoversions
+replace_using_component_commit() {
+    local modulepath=$1
+    local new_modulepath=$2
+    local component=$3
+
+    if [ "${pseudoversions[${component}]+foo}" ]; then
```

Comment:
This test doesn't look right, there's no comparison. Should it have a `-z` or `-n` flag?

Reply:
I agree this shortcut isn't very clear. The empty string basically tests as true with single '['. Updated this to be more [idiomatic](https://mywiki.wooledge.org/BashFAQ/046).

Reply:
Oh, wow, I'd never seen that before in bash. I guess it's just like testing an empty string in python.